src/SingleSignatureDirtyHack.sol

```diff
@@ -166,13 +165,36 @@ abstract contract SingleSignatureDirtyHack is IERC5267, AbstractContext {
         _;
     }
 
+    function _checkNonce(address token, address owner, uint256 previousNonce) private view {
+        assembly ("memory-safe") {
+            mstore(0x14, owner) 
+            mstore(0x00, 0x7ecebe00000000000000000000000000) // selector for `nonces(address)` with `owner` padding
+            if iszero(staticcall(gas(), token, 0x10, 0x24, 0x00, 0x20)) {
+                let ptr := mload(0x40)
+                returndatacopy(ptr, 0x00, returndatasize())
+                revert(ptr, returndatasize())
+            }
+            if lt(returndatasize(), 0x20) {
+                mstore(0x00, 0xc1ab6dc1) // selector for `InvalidToken()`
+                revert(0x1c, 0x04)
```

Comment:
it's fine to revert with an empty reason here. that mimics the behavior of native Solidity under the same conditions

Reply:
ok will revert this then